### PR TITLE
Added features to filter chats, edit messages, add users to group chat, share images as messages, and create dedicated group chats for events

### LINKED ISSUE #2359: [GSoC] Add feature to filter chats, edit messages, add users to group chat, share images as messages, and create dedicated group chats for events
**Is your feature request related to a problem? Please describe.**
Add feature to filter chats, edit messages, add users to group chat, share images as messages, and create dedicated group chats for events

**Describe the solution you'd like**
Add feature to filter chats, edit messages, add users to group chat, share images as messages, and create dedicated group chats for events

**Describe alternatives you've considered**

**Approach to be followed (optional)**
A clear and concise description of approach to be followed.

**Additional context**

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by disha1202**
**What kind of change does this PR introduce?**
Feature

**Issue Number:**

Fixes #2359 #2352 

**Did you add tests for your changes?**

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

## Summary by CodeRabbit

# Release Notes: Chat and Localization Enhancements

## New Features
- Added comprehensive chat functionality across multiple components.
- Introduced organization-specific chat routing.
- Implemented group chat creation and management.
- Enhanced chat filtering options (all, unread, group chats).
- Added ability to create chat when creating events.
- Integrated checkbox for chat creation during event setup.

## Improvements
- Updated internationalization support for multiple languages (English, French, Hindi, Spanish, Chinese).
- Improved chat message handling, including media attachments and message editing.
- Enhanced contact card and chat room user interfaces.
- Added unread message tracking and marking messages as read.
- Refined styling for chat components and user interactions.

## Bug Fixes
- Refined routing for chat components.
- Improved error handling in chat-related mutations and queries.

## Localization
- Added translation keys for chat-related actions across multiple languages.
- Implemented dynamic text rendering based on user language preferences.

### CODE REVIEW

**Comment by palisadoes** [comment]
@disha1202

**Comment by palisadoes** [comment]
@disha1202 

We were having difficulties merging PRs because some of your recent changes created schema instrospection check errors. This PR reverted some of your changes.

https://github.com/PalisadoesFoundation/talawa-admin/pull/2391/files

Please open two PRs for both the API and Admin to restore the feature without the to be simultaneously merged so that the intropection PR errors don't occur.

**Comment by palisadoes** [comment]
> @disha1202
> 
> We were having difficulties merging PRs because some of your recent changes created schema introspection check errors. This PR reverted some of your changes.
> 
> https://github.com/PalisadoesFoundation/talawa-admin/pull/2391/files
> 
> Please open two PRs for both the API and Admin to restore the feature without the to be simultaneously merged so that the intropection PR errors don't occur.

@DMills27 @Kevoniat Please take note.

**Comment by palisadoes** [comment]
@disha1202 Are you still working on this? Please fix the failing tests and conflicting files

**Comment by palisadoes** [comment]
@disha1202 We have made significant changes to the `develop-postgres` branch that there will be major conflicts with this PR when we merge.

**Comment by palisadoes** [comment]
@disha1202 I rebased your PR against the `develop-postgres` branch. We did some changes to the UI/UX color scheme. Please make sure that those are maintained when you refactor the conflicting files.

**Comment by Cioppolo14** [comment]
@disha1202 I am closing this as it seems inactive.

**Comment by palisadoes** [comment]
Reopening. GSoC

**Comment by gurramkarthiknetha** [comment]
If no one is currently working on this issue, I would like to take it on. Please assign it to me.

**Comment by palisadoes** [comment]
Please fix:

1. Any failing tests (Click on the `details` link for more information)
1. Any conflicting files

Make sure CodeRabbit.ai approves your changes

**Comment by Cioppolo14** [comment]
@disha1202 Please fix the failed tests & coderabbit comments.

**Comment by khushipatil1523** [comment]
@palisadoes If no one is working on this issue, can I work on it?

**Comment by palisadoes** [comment]
@disha1202 is working on it

File: src/components/UserPortal/ChatRoom/ChatRoom.tsx [hunk 1]
```diff
@@ -177,6 +251,22 @@
...
+  const handleImageChange = async (
+    e: React.ChangeEvent<HTMLInputElement>,
+  ): Promise<void> => {
+    const file = e.target.files?.[0];
+    if (file) {
+      const base64 = await convertToBase64(file);
+      setAttachment(base64);
+    }
+  };
+
```
[the remark this answers is not in the document]
Comment (disha1202) [reply src/components/UserPortal/ChatRoom/ChatRoom.tsx:269]: Not Required

File: src/components/UserPortal/ChatRoom/ChatRoom.tsx [hunk 2]
```diff
@@ -114,7 +148,23 @@ export default function chatRoom(props: InterfaceChatRoomProps): JSX.Element {
...
+  });
+
+  const [editChatMessage] = useMutation(EDIT_CHAT_MESSAGE, {
+    variables: {
+      messageId: editMessage?._id,
       messageContent: newMessage,
+      chatId: props.selectedContact,
+    },
+  });
+
```
[the remark this answers is not in the document]
Comment (disha1202) [reply src/components/UserPortal/ChatRoom/ChatRoom.tsx:163]: Not required

File: src/components/GroupChatDetails/GroupChatDetails.test.tsx [hunk 3]
```diff
@@ -0,0 +1,603 @@
...
+      },
+      { timeout: 5000 },
+    );
+
+    await act(async () => {
+      fireEvent.click(await screen.findByTestId('addUserBtn'));
+    });
+
+    await wait(10000);
+  });
```
[the remark this answers is not in the document]
Comment (disha1202) [reply src/components/GroupChatDetails/GroupChatDetails.test.tsx:563]: Not required

File: src/components/UserPortal/CreateGroupChat/CreateGroupChat.module.css [hunk 4]
```diff
@@ -7,3 +7,32 @@
...
+  position: relative;
+  top: -40px;
+  left: 40px;
+}
+
+.chatImage {
+  height: 120px;
+  border-radius: 100%;
+  width: 120px;
+}
```
[the remark this answers is not in the document]
Comment (disha1202) [reply src/components/UserPortal/CreateGroupChat/CreateGroupChat.module.css:38]: Not required

**Comment by palisadoes** [comment]
@disha1202 Welcome back. Can this be merged?

**Comment by disha1202** [comment]
> @disha1202 Welcome back. Can this be merged?

Hi @palisadoes 
Yes, this PR is ready to merge but before that we need to merge PRs in talawa api in the following order
1. https://github.com/PalisadoesFoundation/talawa-api/pull/2611
2. https://github.com/PalisadoesFoundation/talawa-api/pull/2625

**Comment by palisadoes** [comment]
Please fix the conflicting files here:
- https://github.com/PalisadoesFoundation/talawa-api/pull/2625

**Review by palisadoes — APPROVED** [review]

**Pull request merged**